lib/luaT/luaT.c

```diff
@@ -497,27 +497,77 @@ int luaT_classmodulename(const char *tname, char *module_name)
   return tname[n] == '.';
 }
 
-/* Lua only functions */
-int luaT_lua_newmetatable(lua_State *L)
+/* Method for determining name of class module */
+const char* luaT_classname(const char *tname)
```

Comment:
I think we should clarify how `luaT_classrootname` and `luaT_classmodulename` should behave for nested modules. `luaT_classrootname` should probably behave just like your `luaT_classname`.